bin/install_swiftpm_dependencies

```diff
@@ -109,16 +160,27 @@ if [[ -d "${XCWORKSPACE_PATH}" ]]; then
   # (despite the help page of `xcodebuild` suggesting that it should work without `-scheme`). Since the dependency resolution doesn't really depend on the scheme
   # and we don't want to have to provide or guess it, using `-list` instead stops making `xcodebuild` complain about `-workspace` not being used in conjunction
   # with `-scheme` (even if in practice we don't care about the scheme list it returns)
-  xcodebuild -workspace "${XCWORKSPACE_PATH}" -resolvePackageDependencies -onlyUsePackageVersionsFromResolvedFile -list
+  xcodebuild -workspace "${XCWORKSPACE_PATH}" -resolvePackageDependencies -skipPackageUpdates -list
 elif [[ -d "${XCODEPROJ_PATH}" ]]; then
   echo "~~~ Resolving Swift Packages with \`xcodebuild\`"
   echo "Using -project \"${XCODEPROJ_PATH}\""
-  xcodebuild -project "${XCODEPROJ_PATH}" -resolvePackageDependencies -onlyUsePackageVersionsFromResolvedFile
+  xcodebuild -project "${XCODEPROJ_PATH}" -resolvePackageDependencies -skipPackageUpdates
 elif [[ "${USE_SPM}" == "true" ]]; then
   echo "~~~ Resolving packages with \`swift package\`"
   swift package resolve
 fi
 
+# If STRICT_PACKAGE_CHECK is enabled, check to see if the `Package.resolved` has uncommitted changes
+if [[ "$STRICT_PACKAGE_CHECK" == "true" ]]; then
+  CHANGES=$(git status --porcelain -- "$PACKAGE_RESOLVED_LOCATION")
+
+  if [[ -n "$CHANGES" ]]; then
+    echo "Uncommitted changes detected in $PACKAGE_RESOLVED_LOCATION:"
+    echo "$CHANGES"
+    exit 1
+  fi
+fi
+
```

Comment:
Given the discussion above that we want idempotence on CI, using `-onlyUsePackageVersionsFromResolvedFile` already guarantees that the `Package.resolved` will not be modified by CI, so this new code won't work in this context as we'll keep using `-onlyUsePackageVersionsFromResolvedFile`, and should thus be deleted.